airflow/cli/commands/user_command.py

```diff
@@ -112,6 +112,8 @@ def users_delete(args):
     """Deletes user from DB."""
     user = _find_user(args)
 
+    user.roles = []  # Clear foreign keys on this user first.
```

Comment:
```suggestion
    user.roles.clear()
```

Do you think using `clear` could be better? It would use the existing reference to the object to clear the list and show our intent directly rather than creating a new empty list and assigning it. We won't need the additional comment in that case. What are your thoughts?

I don't have a strong opinion though :)

Reply:
I agree, it's cleaner and can do the job